web/modules/custom/joinup_invite/joinup_invite.behat.inc

```diff
@@ -74,11 +142,103 @@ class JoinupInviteSubContext extends DrupalSubContextBase implements DrupalSubCo
         'invitation_bundle' => 'discussion',
         'message_template' => InviteToDiscussionForm::TEMPLATE_DISCUSSION_INVITE,
       ],
+      'collection' => [
+        'invitation_bundle' => 'group',
+        'message_template' => InviteToGroupForm::TEMPLATE_GROUP_INVITE,
+      ],
     ];
 
     return $mapping[$entity_type];
   }
 
+  /**
+   * Loads an unchanged invitation.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The entity related to the invitation.
+   * @param \Drupal\Core\Session\AccountInterface $recipient
+   *   The recipient related to the invitation.
+   * @param string $bundle
+   *   The bundle of the invitation.
+   *
+   * @return \Drupal\joinup_invite\Entity\InvitationInterface
+   *   The invitation object.
+   *
+   * @throws \Exception
+   *   Thrown when an invitation is not found.
+   */
+  protected function loadInvitationUnchangedByEntityAndUser(ContentEntityInterface $entity, AccountInterface $recipient, string $bundle = ''): ?InvitationInterface {
+    $storage = \Drupal::entityTypeManager()->getStorage('invitation');
+    $query = $storage->getQuery();
+    $query->condition('entity_type', $entity->getEntityTypeId())
+      ->condition('entity_id', $entity->id())
+      ->condition('recipient_id', $recipient->id())
+      ->condition('bundle', $bundle);
+    $results = $query->execute();
+    if (empty($results)) {
+      throw new Exception('No invitation was found.');
+    }
+
+    $iid = reset($results);
+    /** @var \Drupal\joinup_invite\Entity\InvitationInterface $invitation */
+    $invitation = $storage->loadUnchanged($iid);
+    return $invitation;
+  }
+
+  /**
+   * Searches the accept/reject invitation link from an email and clicks to it.
+   *
+   * @param string $action
+   *   The action to be taken. Either accept or reject.
+   * @param string $user
+   *   The user name.
+   *
+   * @throws \Exception
+   *   Thrown when a user, the email, or the link is not.
+   *
+   * @When I click the :action invitation link from the last email sent to :user
+   */
+  public function clickDeleteConfirmationLink(string $action, string $user): void {
+    $user = $this->getUserByName($user);
+    $user_email = $user->getEmail();
+
+    \Drupal::state()->resetCache();
+    $mails = \Drupal::state()->get('system.test_mail_collector');
+    if (empty($mails)) {
+      throw new Exception('No mail was sent.');
```

Comment:
Always use the full namespace when referring top-level-namespace classes. We are using `\Drupal`, not `Drupal`. So, s/`Exception`/`\Exception`.